package/src/languageService/kustoLanguageService.ts

```diff
@@ -1046,12 +1046,11 @@ class KustoLanguageService implements LanguageService {
         }
 
         // since V2 doesn't support control commands, we're initializing V1 intellisense for both cases and we'll going to use V1 intellisense for control commands.
-        return new Promise((resolve, reject) => {
-            const kustoJsSchema = schema ? KustoLanguageService.convertToKustoJsSchema(schema) : undefined;
-            this._kustoJsSchema = kustoJsSchema;
-            this.createRulesProvider(kustoJsSchema, schema.clusterType);
-            resolve(undefined);
-        });
+
+        const kustoJsSchema = schema ? KustoLanguageService.convertToKustoJsSchema(schema) : undefined;
+        this._kustoJsSchema = kustoJsSchema;
+        this.createRulesProvider(kustoJsSchema, schema.clusterType);
+        return Promise.resolve();
```

Comment:
notice this does slightly change the order of this. When it is inside the promise, it's a microtask that is added to the end of the current eventloop run.

Reply:
why does it need to be a promise at all?

Reply:
> notice this does slightly change the order of this. When it is inside the promise, it's a microtask that is added to the end of the current eventloop run.

Yeah, my assumption that if anything actually happens before the microtask runs it could cause issues. For example, if this was called twice in a row synchronously, the order wouldn't make sense.

> why does it need to be a promise at all?

I'm * guessing * that it doesn't actually need to be one, and that this was written to fulfill the main thread version of it's interface, in which everything returns a promise.